### PR TITLE
ES6ify and Migrate DomainSearch to use isUpgradeable

### PR DESCRIPTION
**Pull request by Tug**
Part of #11328

This refactor of `DomainSearch` is part of an effort to remove `SitesList`.
In #12059, the `isUpgradeable` selector was introduced which allowed us to replace checks on `site.isUpgradeable()`.

### Testing Instructions
Boot the branch locally and visit
- `http://calypso.localhost:3000/domains/add/:siteSlug`
- `http://calypso.localhost:3000/domains/add/suggestion/hello+world/:siteSlug`

to check that there is no regression on the domain search component.
Try switching site while being on each page.

### Reviews
- [x] Code
- [x] Product

### CODE REVIEW

File: client/my-sites/upgrades/domain-search/domain-search.jsx [hunk 1]
```diff
@@ -149,13 +144,19 @@ var DomainSearch = React.createClass( {
...
 
-module.exports = connect(
+export default connect(
 	( state ) => ( {
-		domainsWithPlansOnly: currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+		selectedSite: getSelectedSite( state ),
+		selectedSiteId: getSelectedSiteId( state ),
+		selectedSiteSlug: getSelectedSiteSlug( state ),
+		domainsWithPlansOnly: currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY ),
+		isSiteUpgradeable: siteId => isSiteUpgradeable( state, siteId ),
```
Comment (tyxla) [inline client/my-sites/upgrades/domain-search/domain-search.jsx]: One other thing: can we cache the `isSiteUpgradeable`, so they would be created only once for each `siteId`, instead of being created every time?

File: client/my-sites/upgrades/domain-search/domain-search.jsx [hunk 2]
```diff
@@ -149,13 +144,19 @@ var DomainSearch = React.createClass( {
...
+		productsList: state.productsList.items,
```
Comment (Tug) [inline client/my-sites/upgrades/domain-search/domain-search.jsx]: Will do in another PR ;)

File: client/my-sites/upgrades/domain-search/domain-search.jsx [hunk 3]
```diff
@@ -139,7 +134,7 @@ var DomainSearch = React.createClass( {
 							selectedSite={ selectedSite }
```
Comment (Tug) [inline client/my-sites/upgrades/domain-search/domain-search.jsx]: That doesn't seem like a real case scenario, a user would not clear his local storage without reloading and expect the app to work.
We should however make sure that loading any page with an empty localStorage won't break this component. I'll do some testings

Comment (Tug) [inline client/my-sites/upgrades/domain-search/domain-search.jsx]: It seems `selectedSite` can indeed be `null` when we switch site (don't even need to clear the localStorage). I have updated propTypes to reflect that.
I think the work on site switching with an empty state is a much deeper issue and probably already breaks in many cases currently.
Here we have upgraded the component while keeping the same behavior which is already a step forward.

File: client/my-sites/upgrades/domain-search/domain-search.jsx [hunk 4]
```diff
@@ -1,89 +1,83 @@
...
-			page.redirect( '/domains/add' );
-		}
-	},
-
-	handleDomainsAvailabilityChange: function( isAvailable ) {
+import isSiteUpgradeable from 'state/selectors/is-site-upgradeable';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import QueryProductsList from 'components/data/query-products-list';
+
+const analytics = analyticsMixin( 'registerDomain' );
```
Comment (gziolo) [inline client/my-sites/upgrades/domain-search/domain-search.jsx]: Interesting workaround :)
